js/graviex.js

```diff
@@ -0,0 +1,991 @@
+        if ('side' in trade) {
+            if (side === 'buy') {
+                takerOrMaker = 'maker';
+            } else {
+                takerOrMaker = 'taker';
+            }
+        }
+        return this.safeTrade ({
+            'info': trade,
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'symbol': symbol,
+            'id': id,
+            'order': orderId,
+            'type': undefined,
+            'side': side,
+            'takerOrMaker': takerOrMaker,
+            'price': price,
+            'funds': funds,
+            'amount': amount,
+            'cost': undefined,
+            'fee': undefined,
+        }, market);
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const request = {
+            'market': market['id'],
+        };
+        const defaultType = 'spot';
+        const type = this.safeString (params, 'type', defaultType);
+        const query = this.omit (params, 'type');
+        let defaultMethod = undefined;
+        if (type === 'simple') {
+            defaultMethod = 'publicGetTradesSimple';
+        } else {
+            defaultMethod = 'publicGetTrades';
+        }
+        const method = this.safeString (this.options, 'fetchTradesMethod', defaultMethod);
+        if (limit !== undefined) {
+            request['limit'] = limit; // default: 500
+        }
+        const response = await this[method] (this.extend (request, query));
+        // [
+        //     {
+        //         "id": 14473952,
+        //         "at": 1643681026,
+        //         "price": "0.00000056",
+        //         "volume": "0.2",
+        //         "funds": "0.000000112",
+        //         "market": "giobtc",
+        //         "created_at": "2022-02-01T05:03:46+03:00",
+        //         "side": "sell"
+        //     },
+        //     {
+        //         "id": 14473615,
+        //         "at": 1643670349,
+        //         "price": "0.00000059",
+        //         "volume": "515.0136",
+        //         "funds": "0.000303858024",
+        //         "market": "giobtc",
+        //         "created_at": "2022-02-01T02:05:49+03:00",
+        //         "side": "buy"
+        //     },
+        // ],
+        return this.parseTrades (response, market, since, limit);
+    }
+
+    parseOHLCV (ohlcv, market = undefined) {
+        return [
+            this.safeInteger (ohlcv, 0),
+            this.safeNumber (ohlcv, 1),
+            this.safeNumber (ohlcv, 2),
+            this.safeNumber (ohlcv, 3),
+            this.safeNumber (ohlcv, 4),
+            this.safeNumber (ohlcv, 5),
+        ];
+    }
+
+    async fetchOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        const market = this.market (symbol);
+        const defaultLimit = 50;
```

Comment:
i don't think we should sent a defaultLimit, if the user does not specify it and it is optional we should not send anything

Reply:
Okay. Fixed in the latest commit https://github.com/ccxt/ccxt/pull/11697/commits/162b106bdb42084917d6187a3a60fb6e0f55e7d1